library/Icingadb/Util/PerfData.php

```diff
@@ -307,8 +451,8 @@ protected function parse()
         $parts = explode(';', $this->perfdataValue);
 
         $matches = array();
-        if (preg_match('@^(-?\d+(\.\d+)?)([a-zA-Z%°]{1,2})$@u', $parts[0], $matches)) {
-            $this->unit = strtolower($matches[3]);
+        if (preg_match('@^(-?\d+(\.\d+)?)([a-zA-Z%°]{1,3})$@u', $parts[0], $matches)) {
+            $this->unit = $matches[3];
             $this->value = self::convert($matches[1], $this->unit);
```

Comment:
I'm not sure if values need to be converted here still. If we expect the input data as being normalized, there's no need to. Also, doesn't the formatting in method `format()` expect the normalized values from Icinga?

Reply:
Ohh, you're right. I wonder if this is also the case for the thresholds, so are the thresholds normalized as well or does this only affect the plugin output?

Reply:
That is a question for @Al2Klimov :wink:

Reply:
Reading about _plugin output normalization_ the first time. 😉

And – regarding the thresholds:

https://github.com/Icinga/icinga2/blob/63b53bfda92930b2332035860082dbcaa5cff9e7/test/icinga-perfdata.cpp#L46-L53

Reply:
> Reading about _plugin output normalization_ the first time. 😉

My mistake.